packages/driver/src/cypress/source_map_utils.ts

```diff
@@ -1,39 +1,53 @@
 import _ from 'lodash'
 import { SourceMapConsumer } from 'source-map'
-import Promise from 'bluebird'
+import Bluebird from 'bluebird'
 
+import type { BasicSourceMapConsumer } from 'source-map'
 // @ts-ignore
 import mappingsWasm from 'source-map/lib/mappings.wasm'
 
 import $utils from './utils'
+import type BluebirdPromise from 'bluebird'
 
 const sourceMapExtractionRegex = /\/\/\s*[@#]\s*sourceMappingURL\s*=\s*(data:[^\s]*)/g
 const regexDataUrl = /data:[^;\n]+(?:;charset=[^;\n]+)?;base64,([a-zA-Z0-9+/]+={0,2})/ // matches data urls
 
-let sourceMapConsumers = {}
+let sourceMapConsumers: Record<string, BasicSourceMapConsumer> = {}
 
-const initializeSourceMapConsumer = (file, sourceMap) => {
-  if (!sourceMap) return Promise.resolve(null)
+const initializeSourceMapConsumer = (script, sourceMap): BluebirdPromise<BasicSourceMapConsumer | null> => {
+  if (!sourceMap) return Bluebird.resolve(null)
 
   // @ts-ignore
   SourceMapConsumer.initialize({
     'lib/mappings.wasm': mappingsWasm,
   })
 
-  return Promise.resolve(new SourceMapConsumer(sourceMap)).then((consumer) => {
-    sourceMapConsumers[file.fullyQualifiedUrl] = consumer
+  return Bluebird.resolve(new SourceMapConsumer(sourceMap))
+  .then((consumer) => {
+    sourceMapConsumers[script.fullyQualifiedUrl] = consumer
 
     return consumer
   })
 }
 
-const extractSourceMap = (file, fileContents) => {
-  let sourceMapMatch = fileContents.match(sourceMapExtractionRegex)
+const extractSourceMap = (fileContents) => {
+  let dataUrlMatch
 
-  if (!sourceMapMatch) return null
+  try {
+    let sourceMapMatch = fileContents.match(sourceMapExtractionRegex)
+
+    if (!sourceMapMatch) return null
 
-  const url = _.last(sourceMapMatch) as any
-  const dataUrlMatch = url.match(regexDataUrl)
+    const url = _.last(sourceMapMatch) as any
+
+    dataUrlMatch = url.match(regexDataUrl)
+  } catch (err) {
+    // ignore unable to match regex. there's nothing we
+    // can do about it and we don't want to thrown an exception
+    if (err.message === 'Maximum call stack size exceeded') return null
```

Comment:
Might be worth checking if this error is the same in FF and WebKit

Reply:
I was going to suggest checking the error type instead, but it looks like the error type is inconsistent (InternalError in Firefox; RangeError in Chrome and Safari.), but the error *message* is the same across browsers.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Too_much_recursion